service/src/test/java/uk/gov/hmcts/reform/fpl/service/orders/generator/DocumentMergerTest.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.fpl.service.orders.generator;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.DocumentDownloadService;
+import uk.gov.hmcts.reform.fpl.service.docmosis.DocumentConversionService;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readBytes;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocumentReference;
+
+@ExtendWith(MockitoExtension.class)
+class DocumentMergerTest {
+    @Mock
+    private DocumentConversionService documentConversionService;
+
+    @Mock
+    private DocumentDownloadService documentDownloadService;
+
+    @InjectMocks
+    private DocumentMerger underTest;
+
+    private static final String ORIGINAL_FILENAME = "document1.pdf";
+    private static final String ADDITIONAL_FILENAME = "document2.pdf";
+
+    private static final DocmosisDocument DOCMOSIS_DOCUMENT = DocmosisDocument.builder()
+        .documentTitle(ORIGINAL_FILENAME).bytes(new byte[]{1}).build();
+
+    private static final DocumentReference DOCUMENT_REFERENCE = testDocumentReference(ADDITIONAL_FILENAME);
+
+    @Test
+    void shouldMergeTheDocuments() {
+        final byte[] originalDocument = readBytes("documents/document1.pdf");
```

Comment:
Do we need the additional documents? Can we not just have some dummy byte arrays like you do with the additionalDocumentBytes?

Reply:
I'm using dummy daya for downloading the additional document from `DocumentReference` type. `convertToPdf`method should return the actual file data, which will then be merged using pdf box utility class